NewHorizons/Builder/Body/SingularityBuilder.cs

```diff
@@ -95,51 +91,57 @@ public static void Make(GameObject go, Sector sector, OWRigidbody OWRB, PlanetCo
+            if (!string.IsNullOrEmpty(pairedSingularity))
             {
-                if (_singularitiesByID.TryGetValue(pairedSingularity, out var pairedSingularityGO))
+                if (polarity)
                 {
-                    switch (polarity)
-                    {
-                        case true:
-                            PairSingularities(uniqueID, pairedSingularity, newSingularity, pairedSingularityGO);
-                            break;
-                        case false:
-                            PairSingularities(pairedSingularity, uniqueID, pairedSingularityGO, newSingularity);
-                            break;
-                    }
+                    _pairsToLink.Add((uniqueID, pairedSingularity));
+                }
+                else
+                {
+                    _pairsToLink.Add((pairedSingularity, uniqueID));
                 }
             }
+
         }
 
-        public static void PairSingularities(string blackHoleID, string whiteHoleID, GameObject blackHole, GameObject whiteHole)
+        public static void PairAllSingularities()
         {
-            InitPrefabs();
-
-            if (blackHole == null || whiteHole == null) return;
-
-            Logger.LogVerbose($"Pairing singularities [{blackHoleID}], [{whiteHoleID}]");
-
-            var whiteHoleVolume = whiteHole.GetComponentInChildren<WhiteHoleVolume>();
-            var blackHoleVolume = blackHole.GetComponentInChildren<BlackHoleVolume>();
-
-            if (whiteHoleVolume == null || blackHoleVolume == null)
+            foreach (var pair in _pairsToLink)
             {
-                Logger.LogWarning($"[{blackHoleID}] and [{whiteHoleID}] do not have compatible polarities");
-                return;
+                var (blackHoleID, whiteHoleID) = pair;
+                if (!_singularitiesByID.TryGetValue(blackHoleID, out GameObject blackHole))
+                {
+                    Logger.LogWarning($"Black hole [{blackHoleID}] is missing.");
+                    break;
+                }
+                if (!_singularitiesByID.TryGetValue(whiteHoleID, out GameObject whiteHole))
+                {
+                    Logger.LogWarning($"White hole [{whiteHoleID}] is missing.");
+                    break;
+                }
+                var whiteHoleVolume = whiteHole.GetComponentInChildren<WhiteHoleVolume>();
+                var blackHoleVolume = blackHole.GetComponentInChildren<BlackHoleVolume>();
+                if (whiteHoleVolume == null || blackHoleVolume == null)
+                {
+                    Logger.LogWarning($"Singularities [{blackHoleID}] and [{whiteHoleID}] do not have compatible polarities.");
+                    break;
+                }
+                if (blackHoleVolume._whiteHole != null && blackHoleVolume._whiteHole != whiteHoleVolume)
+                {
+                    Logger.LogWarning($"Black hole [{blackHoleID}] has already been linked!");
```

Comment:
What if we just let multiple black holes lead to a single white hole? Could be interesting. Would it break anything?

Reply:
Technically already worked, if you set pairedSingularity on the black holes but not the white hole, but it would fail to pair if the while hole was created after. But after this change, it'll always work.

Reply:
I read wrong but also nice very nice